src/compiler/program.ts

```diff
@@ -84,7 +83,7 @@ namespace ts {
             let text: string | undefined;
             try {
                 performance.mark("beforeIORead");
-                text = system.readFile(fileName, options.charset);
+                text = host.readFile(fileName);

```

Comment:
what is the difference between system and host here and in the next change?

Reply:
host is compilerHost being created. May be I should name it compilerHost to make it more clear. The change here was to use compilerHost's read so if someone overriding compilerHost's read method (eg tsbuild's new caching logic) can kick in place.

Reply:
Oh and system is `ts.sys like interface`